pallets/moonbeam-lazy-migrations/src/foreign_asset.rs

```diff
@@ -171,14 +178,29 @@ where
 					<T as pallet_assets::Config>::Currency::unreserve(&owner, approval.deposit);
 
 					MIGRATING_FOREIGN_ASSETS::using_once(&mut true, || {
-						pallet_moonbeam_foreign_assets::Pallet::<T>::approve(
+						let address: H160 = owner.clone().into();
+
+						// Temporarily remove metadata
```

Comment:
Could you expand on why this is needed?

Reply:
When migrating approvals, some calls require calling the contract with the approval owner as origin (a smart-contract in this case), which is not allowed, only EOA can perform external calls. In the case of this migration this must be allowed, and we intentionally remove and re-add the metadata.